src/components/tab/SearchInput.vue

```diff
@@ -0,0 +1,308 @@
+				if (response.status === 200) {
+					this.state = 'loading'
+					this.processWorkPackages(response.data)
+				} else {
+					this.checkStatusCode(response.status)
+				}
+			} else {
+				this.state = 'ok'
+			}
+		},
+		checkStatusCode(statusCode) {
+			if (statusCode === 401) {
+				this.state = 'no-token'
+			} else if (statusCode === 400) {
+				this.state = 'error'
+			} else {
+				this.state = 'error'
+			}
+		},
+		replaceHrefToGetId(href, href2 = null) {
+			if (href2 !== null) {
+				return href
+					? href.replace(/.*\//, '')
+					: href2
+						? href2.replace(/.*\//, '')
+						: null
+			}
+			return href
+				? href.replace(/.*\//, '')
+				: null
+		},
+		async processWorkPackages(workPackages) {
+			if (workPackages.length === 0) {
+				this.state = 'empty'
+				return
+			}
+			for (let i = 0; i < workPackages.length; i++) {
+				const statusId = this.replaceHrefToGetId(workPackages[i]._links.status.href)
+				const typeId = this.replaceHrefToGetId(workPackages[i]._links.type.href)
+				const userId = this.replaceHrefToGetId(workPackages[i]._links?.assignee?.href, workPackages[i]._links?.author?.href)
+				const userName = workPackages[i]._links?.assignee?.title
+					? workPackages[i]._links.assignee.title
+					: workPackages[i]._links?.author?.title
+						? workPackages[i]._links.author.title
+						: null
+				const avatar = await this.getUserAvatar(userId, userName)
+				const statusColor = await this.processWPStatus(statusId)
+				const typeColor = await this.processWPType(typeId)
+				const found = this.searchResults.some(el => el.id === workPackages[i].id)
+				if (!found) {
+					this.searchResults.push({
+						id: workPackages[i].id,
+						subject: workPackages[i].subject,
+						project: workPackages[i]._links.project.title,
+						statusTitle: workPackages[i]._links.status.title,
+						typeTitle: workPackages[i]._links.type.title,
+						assignee: workPackages[i]._links.assignee.title,
+						statusCol: statusColor,
+						typeCol: typeColor,
+						picture: avatar,
+					})
+				}
+			}
+		},
+		async processWPStatus(id) {
+			const url = generateUrl('/apps/integration_openproject/statuses/' + id)
+			const req = {}
+			const response = await axios.get(url, req)
+			this.state = 'loading'
+			if (response.status === 200) {
+				this.state = 'ok'
+				return response.data.color
+			}
+			this.checkStatusCode(response.status)
+			return ''
+		},
+		async processWPType(id) {
+			const url = generateUrl('/apps/integration_openproject/types/' + id)
+			const req = {}
+			const response = await axios.get(url, req)
+			this.state = 'loading'
+			if (response.status === 200) {
+				this.state = 'ok'
+				return response.data.color
+			}
+			this.checkStatusCode(response.status)
+			return ''
+		},
+		async getUserAvatar(userId, userName) {
+			const url = generateUrl('/apps/integration_openproject/avatar?')
+				+ encodeURIComponent('userId') + '=' + userId + '&' + encodeURIComponent('userName') + '=' + userName
```

Comment:
generateUrl accepts parameters
I think something like this should work:
```suggestion
			const url = generateUrl('/apps/integration_openproject/avatar', {'userId': userId, 'userName': userName )
```
but I haven't tried it